wp_api/src/lib.rs

```diff
@@ -137,6 +137,27 @@ pub enum JsonValue {
     Object(HashMap<String, JsonValue>),
 }
 
+uniffi::custom_newtype!(WpResponseString, Option<String>);
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(from = "BoolOrString")]
+pub struct WpResponseString(pub Option<String>);
+
+impl From<BoolOrString> for WpResponseString {
+    fn from(value: BoolOrString) -> Self {
+        match value {
+            BoolOrString::Bool(_) => Self(None),
```

Comment:
Return error if a `true` value is returned?

Reply:
We could do that, but if this happens, it'll be due to a bug in the API and we probably don't need to propagate that. 🤔

Reply:
What do you think @crazytonyli @jkmassel?

Reply:
Yes it'd be a bug in the API. It'd be an API bug too if we got `"avatar_urls": ["...", "..."]`, which would cause a parsing error. I think for consistency, we should throw a parsing error here, too, when receiving a `true` value, because it's an unexpected value, in the same way a list is an unexpected value for `avatar_urls`.

I don't mind it too much, though. That's why I approved the PR...

Reply:
I could be wrong, but my thinking is that the issue this PR fixes, an object returning as `[]` etc are all expected behavior. I agree with you that they should be considered bugs, but they are not. So, we have to handle them - otherwise we can't use the endpoints at all.

On the other hand, returning `true` when a `String` is expected doesn't make any sense. `false` does in a weird way - although it really should be `null`. That's why I thought it's better not to cause any client side errors for this.

I don't feel strongly about it, so happy to go with whichever.

Reply:
@crazytonyli I wanted to get this merged, so I implemented the `try_from` approach as well in 395c8e25377e65fd49480c36fb05fc441a4e1256. Let me know what you think!